setup.py

```diff
@@ -72,13 +72,54 @@ def build_extension(self, ext):
     raise RuntimeError('Unable to determine version.')
 
 
-install_requires = ['attrs>=17.3.0', 'chardet>=2.0,<4.0',
-                    'multidict>=4.0,<5.0',
-                    'async_timeout>=3.0,<4.0',
-                    'yarl>=1.0,<2.0']
-
-if sys.version_info < (3, 7):
-    install_requires.append('idna-ssl>=1.0')
+def get_environment_marker_support_level():
+    """
+    Copied from https://github.com/pytest-dev/pytest/blob/master/setup.py
+
+    Tests how well setuptools supports PEP-426 environment marker.
+    The first known release to support it is 0.7 (and the earliest on PyPI seems to be 0.7.2
+    so we're using that), see: https://setuptools.readthedocs.io/en/latest/history.html#id350
+    The support is later enhanced to allow direct conditional inclusions inside install_requires,
+    which is now recommended by setuptools. It first appeared in 36.2.0, went broken with 36.2.1, and
+    again worked since 36.2.2, so we're using that. See:
+    https://setuptools.readthedocs.io/en/latest/history.html#v36-2-2
+    https://github.com/pypa/setuptools/issues/1099
+    References:
+    * https://wheel.readthedocs.io/en/latest/index.html#defining-conditional-dependencies
+    * https://www.python.org/dev/peps/pep-0426/#environment-markers
+    * https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies
+    """
+    try:
+        version = pkg_resources.parse_version(setuptools.__version__)
+        if version >= pkg_resources.parse_version("36.2.2"):
+            return 2
+        if version >= pkg_resources.parse_version("0.7.2"):
```

Comment:
Should we care about such old versions? I think we shouldn't.

Reply:
OK